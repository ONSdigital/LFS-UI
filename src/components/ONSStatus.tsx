import React, { Component } from 'react';

interface Props {
    label: string,
    id?: string,
    small: boolean
    status: string
}

export class ONSStatus extends Component <Props, {}>{
    constructor(props : Props) {
        super(props);
    }
    render() {
        let className = "status status--" + this.props.status + (this.props.small ? " status--small": "");
        return(
            <span className={className}>{this.props.label}</span>
        )
    }
}
