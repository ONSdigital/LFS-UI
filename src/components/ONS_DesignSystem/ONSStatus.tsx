import React, {Component} from 'react';

interface Props {
    label: string,
    small: boolean
    status: string
}

export class ONSStatus extends Component <Props, {}>{

    render() {
        let className = "status status--" + this.props.status + (this.props.small ? " status--small": "");
        return(
            <span className={className}>{this.props.label}</span>
        )
    }
}
