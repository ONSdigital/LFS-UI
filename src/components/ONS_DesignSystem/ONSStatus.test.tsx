import React from 'react';
import { shallow, mount } from 'enzyme';
import {ONSStatus} from "./ONSStatus";
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

describe("ONS Status Test", () => {
    Enzyme.configure({ adapter: new Adapter() })

    const statusProps = {
        label: "success",
        small: true,
        status: "success"
    }

    const smallStatusProps = {
        label: "error",
        small: false,
        status: "error"
    }

    function wrapper(render: any, props: any) {
        return render(
            <ONSStatus  label={props.label}
                        small={props.small}
                        status={props.status}>
            </ONSStatus>
        )
    }
    
    it("should render correctly", () => expect(wrapper(shallow, statusProps).exists()).toEqual(true));
    
    it('displays a small status', () => {
        expect(wrapper(mount, smallStatusProps).find('span').hasClass('status--error')).toEqual(true)
    })
})