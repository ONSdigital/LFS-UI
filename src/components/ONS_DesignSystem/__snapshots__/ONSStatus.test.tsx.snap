// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Status Test matches Snapshot 1`] = `ShallowWrapper {}`;
